src/utils/request/onUnhandledRequest.ts

```diff
@@ -46,18 +48,23 @@ function groupHandlersByType(handlers: RequestHandler[]) {
   )
 }
 
-type RequestHandlerSuggestionList = [number, RequestHandler][]
-type ScoreGetterFn = (request: MockedRequest, handler: RequestHandler) => number
+type RequestHandlerSuggestion = [number, RequestHandler]
 
-function getScoreForRestHandler(): ScoreGetterFn {
+type ScoreGetterFn<RequestHandlerType extends RequestHandler> = (
+  request: MockedRequest,
+  handler: RequestHandlerType,
+) => number
+
+function getRestHandlerScore(): ScoreGetterFn<RestHandler> {
   return (request, handler) => {
     const { path, method } = handler.info
 
-    if (path instanceof RegExp) {
+    if (path instanceof RegExp || method instanceof RegExp) {
       return Infinity
     }
 
     const hasSameMethod = isStringEqual(request.method, method)
+
```

Comment:
Or perhaps change this to something like:
```suggestion
    if (path instanceof RegExp) {
      return Infinity
    }

    const hasSameMethod = method instanceof RegExp || isStringEqual(request.method, method)

```

Reply:
The `method` being a regular expression does not guarantee that it's the _same_ method. Originally, I've done this:

```js
const hasSameMetod = method instanceof RegExp
  ? method.test(request.method)
  : isStringEqual(request.method, method)
```

While this does the method matching, it has an unintended side-effect: this handler being included in the suggestions. You see, when we suggest what handlers you might've meant (i.e. in the case of a typo), we print the _handler's header_. That's a string that represents the handler's resource path, effectively:

- Method + URL for REST handlers;
- Operation Type + Operation Name for GraphQL handlers.

If the handler has `RegExp` as its method, the printed suggestion will look like this:

```
Did you mean to request "/^GE http://localhost/api"?
```

That's not a useful suggestion. In fact, due to the nature of regular expressions being dynamic in nature, any suggestion including RegExp becomes quite useless. Partially, because such request handler's "header" string is not static, but mainly because the users can't really understand what request via are suggesting (RegExp matches a subset of requests). 

This was my main reasoning behind ignoring regular expressions entirely when looking up the handler suggestions. Do you see any potential issues with this that I've overlooked?

Reply:
Shortly, my main concern with dynamic methods/operation names is that their string representation is not useful for the user to suggest. It's insufficient to make a good conclusion over what might've been missed, or what should be corrected.

Reply:
> This was my main reasoning behind ignoring regular expressions entirely when looking up the handler suggestions. Do you see any potential issues with this that I've overlooked?

Not at all. In fact, I would argue that methods are finite and they should be limited to the existing ones.
If a user wants to perform the same handler for multiple verbs/methods, he/she can do one of the following:

```js
const  myHandler = (req, res, ctx) => { /* ... */ };
rest.post('/path', myHandler);
rest.get('/path', myHandler);
rest.put('/path', myHandler);
```
or 
```js
const  myHandler = (req, res, ctx) => {
  const method = req.method.toUpperCase();
  if (method === 'POST') {
    /* ... do POST things ... */
    return res(/* ... */);
  }

  if (method === 'GET') {
    /* ... do GET things ... */
    return res(/* ... */);
  }

  if (method === 'PUT') {
    /* ... do PUT things ... */
    return res(/* ... */);
  }
};
rest.all('/path', myHandler);
```

I'd argue also that `rest.all()` should have lowest precedence when it comes to handlers' execution, since it's less specific.

If I do 

```
rest.all('/path', (req, res, ctx) => { /* ... */ });
rest.post('/path', (req, res, ctx) => { /* ... */ }); // this should be executed as it's more specific
```

but again, that's my opinion 😉

Reply:
Oh and btw, @kettanaito, thanks for your quick reply and action in this 👍

Reply:
Please, just note that this code does not affect how the handlers are run in any way. This is only the logic to _find suggestions_ from the existing handlers. Request handlers have no priority other than their declaration order (which you should utilize if you wish for fall-through and other patterns to apply). This issue (and the fix) are related to how MSW looked up the handlers to suggest when it catches a request that has no handlers. This is a purely DX thing to save from typos and other logical issues.

Reply:
Hmmm I see ... so if I set up `onUnhandledRequest: 'bypass'` it would actually solve my issue 🤦  ... So I happened to find that issue because I was not bypassing unhandled requests 😄  ... thanks for that @kettanaito ...

Reply:
I don't think that using "bypass" would've prevented this issue. The library looks up the suggestions _before_ deciding whether to print them or not (hey, a nice place for improvement!).

If you like my assistance and would want to help me make MSW a better experience, please consider supporting me on [GitHub](https://github.com/sponsors/kettanaito) or our entire team on [Open Collective](https://opencollective.com/mswjs). Thank you!

Reply:
Done! Keep up the good work! 👍

Reply:
Thank you so much, @rmachado-studocu ❤️ You are incredible.